presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java

```diff
@@ -395,21 +396,24 @@ public PrestoReadSupport(List<HiveColumnHandle> columns, MessageType messageType
                     }
                     else {
                         GroupType groupType = parquetType.asGroupType();
-                        switch (groupType.getOriginalType()) {
-                            case LIST:
-                                ParquetJsonColumnConverter listConverter = new ParquetJsonColumnConverter(new ParquetListJsonConverter(groupType.getName(), null, groupType), i);
-                                converters.add(listConverter);
-                                closeableBuilder.add(listConverter);
-                                break;
-                            case MAP:
-                            case MAP_KEY_VALUE: // original versions of Parquet have map and entry swapped
-                                ParquetJsonColumnConverter mapConverter = new ParquetJsonColumnConverter(new ParquetMapJsonConverter(groupType.getName(), null, groupType), i);
-                                converters.add(mapConverter);
-                                closeableBuilder.add(mapConverter);
-                                break;
-                            case UTF8:
-                            case ENUM:
-                                throw new IllegalArgumentException("Group column " + groupType.getName() + " type " + groupType.getOriginalType() + " not supported");
+                        OriginalType originalType = groupType.getOriginalType();
+                        if (originalType == null) { // struct does not have an original type
```

Comment:
This case is handled properly in ``createJsonConverter()`` method, but not here causing NPE.